From 2e14f85910ccc07dedbfe9a2a4c116675d4043fa Mon Sep 17 00:00:00 2001
From: Harlan Lieberman-Berg <hlieberman@setec.io>
Date: Sun, 16 Aug 2015 22:16:23 +0200
Subject: [PATCH 1/4] Make version check for Python urllib3 injection more
 specific.

Because Debian ships a symlinked version of urllib3 in the requests
module, the injection of pyopenssl doesn't work correctly.  Fix that by
avoiding the issue, and don't worry about accidentally breaking another
distro.
---
 acme/acme/client.py | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/acme/acme/client.py b/acme/acme/client.py
index 6906308..584f6f8 100644
--- a/acme/acme/client.py
+++ b/acme/acme/client.py
@@ -8,7 +8,7 @@ from six.moves import http_client  # pylint: disable=import-error
 
 import OpenSSL
 import requests
-import six
+import sys
 import werkzeug
 
 from acme import errors
@@ -19,8 +19,8 @@ from acme import messages
 
 logger = logging.getLogger(__name__)
 
-# https://urllib3.readthedocs.org/en/latest/security.html#insecureplatformwarning
-if six.PY2:
+# Python does not validate certificates by default before version 2.7.9
+if sys.hexversion < 0x20709F0 :
     requests.packages.urllib3.contrib.pyopenssl.inject_into_urllib3()
 
 
